### PR TITLE
feat: Feat allow snapshot tests for serialized addons

### PR DESCRIPTION
**Pull request by lifeart**
implements: https://github.com/NullVoxPopuli/ember-addon-migrator/issues/24

### CODE REVIEW

File: src/addon.js [hunk 1]
```diff
@@ -148,13 +148,16 @@ async function withVersions(packageList) {
   /** @type {Record<string, string>} */
   let mapped = {};
 
-  await Promise.all(
-    packageList.map(async (name) => {
-      let version = await latestVersion(name);
-
-      mapped[name] = `^${version}`;
+  const versions = await Promise.all(
```
Comment (NullVoxPopuli) [inline src/addon.js:151]: Why change this?

File: tests/helpers.ts [hunk 2]
```diff
@@ -6,10 +6,22 @@ import { Project } from 'scenario-tester';
 import { fileURLToPath } from 'url';
 import { expect } from 'vitest';
 
-import { restoreAddon } from './../src/captured-addon/utils/fs-helpers';
+import { captureAddon, restoreAddon } from './../src/captured-addon/utils/fs-helpers';
 
 const __dirname = dirname(fileURLToPath(import.meta.url));
 
+export async function snapshotFrom(addonPath: string) {
```
Comment (NullVoxPopuli) [inline tests/helpers.ts:13]: Why alias captureAddon?

File: tests/helpers.ts [hunk 3]
```diff
@@ -128,3 +140,31 @@ export async function verify(fixtureName: string) {
...
+  await migrate(project);
+
+  const result = await snapshotFrom(project.baseDir);
+
+  const fixturePath = join(__dirname, 'samples', fixtureName);
+
+  const outputPath = join(fixturePath, 'output.json');
+
+  if (fs.existsSync(outputPath)) {
+    const prevResult = JSON.parse(fs.readFileSync(outputPath, 'utf8'));
```
Comment (NullVoxPopuli) [inline tests/helpers.ts:156]: Why so many sync fs operations? Why not async?

File: tests/helpers.ts (hunk 2 above)
Comment (lifeart) [reply tests/helpers.ts:13]: to simplify scope in tests (to not introduce extra import to deal with)

File: tests/helpers.ts (hunk 3 above)
Comment (lifeart) [reply tests/helpers.ts:156]: it's not meaningful for speed in this context, overall test per addon took 120s, and here there is no reason to add async logic

File: tests/helpers.ts (hunk 2 above)
Comment (NullVoxPopuli) [reply tests/helpers.ts:13]: Does it need to be exported? Cal the usage be inlined?

Comment (lifeart) [reply tests/helpers.ts]: oh, my bad, it's not used outside of this file, removed

File: tests/helpers.ts (hunk 3 above)
Comment (NullVoxPopuli) [reply tests/helpers.ts:152]: Ok, sounds good 👍

File: src/addon.js (hunk 1 above)
Comment (NullVoxPopuli) [reply src/addon.js:151]: Gotchya, thanks!

File: tests/helpers.ts (hunk 3 above)
Comment (lifeart) [reply tests/helpers.ts:152]: but, I can add it needed, don't have strong position here

**Review by NullVoxPopuli — APPROVED** [review]

File: tests/helpers.ts (hunk 3 above)
Comment (NullVoxPopuli) [reply tests/helpers.ts:152]: All good. I'm not sure i do either. I've developed a habit of using fs/promises everywhere though. So the introduction of sync fs code is new to this project 🤷

**Pull request merged**